gcloud/dns/test_changes.py

```diff
@@ -0,0 +1,346 @@
+        self.assertEqual(req['query_params'], {'fields': 'id'})
+
+    def test_exists_hit_w_alternate_client(self):
+        PATH = 'projects/%s/managedZones/%s/changes/%s' % (
+            self.PROJECT, self.ZONE_NAME, self.CHANGES_NAME)
+        conn1 = _Connection()
+        client1 = _Client(project=self.PROJECT, connection=conn1)
+        conn2 = _Connection({})
+        client2 = _Client(project=self.PROJECT, connection=conn2)
+        zone = _Zone(client1)
+        changes = self._makeOne(zone)
+        changes.name = self.CHANGES_NAME
+
+        self.assertTrue(changes.exists(client=client2))
+
+        self.assertEqual(len(conn1._requested), 0)
+        self.assertEqual(len(conn2._requested), 1)
+        req = conn2._requested[0]
+        self.assertEqual(req['method'], 'GET')
+        self.assertEqual(req['path'], '/%s' % PATH)
+        self.assertEqual(req['query_params'], {'fields': 'id'})
+
+    def test_reload_w_bound_client(self):
+        PATH = 'projects/%s/managedZones/%s/changes/%s' % (
+            self.PROJECT, self.ZONE_NAME, self.CHANGES_NAME)
+        self._setUpConstants()
+        RESOURCE = self._makeResource()
+        conn = _Connection(RESOURCE)
+        client = _Client(project=self.PROJECT, connection=conn)
+        zone = _Zone(client)
+        changes = self._makeOne(zone)
+        changes.name = self.CHANGES_NAME
+
+        changes.reload()
+
+        self.assertEqual(len(conn._requested), 1)
+        req = conn._requested[0]
+        self.assertEqual(req['method'], 'GET')
+        self.assertEqual(req['path'], '/%s' % PATH)
+        self._verifyResourceProperties(changes, RESOURCE, zone)
+
+    def test_reload_w_alternate_client(self):
+        PATH = 'projects/%s/managedZones/%s/changes/%s' % (
+            self.PROJECT, self.ZONE_NAME, self.CHANGES_NAME)
+        self._setUpConstants()
+        RESOURCE = self._makeResource()
+        conn1 = _Connection()
+        client1 = _Client(project=self.PROJECT, connection=conn1)
+        conn2 = _Connection(RESOURCE)
+        client2 = _Client(project=self.PROJECT, connection=conn2)
+        zone = _Zone(client1)
+        changes = self._makeOne(zone)
+        changes.name = self.CHANGES_NAME
+
+        changes.reload(client=client2)
+
+        self.assertEqual(len(conn1._requested), 0)
+        self.assertEqual(len(conn2._requested), 1)
+        req = conn2._requested[0]
+        self.assertEqual(req['method'], 'GET')
+        self.assertEqual(req['path'], '/%s' % PATH)
+        self._verifyResourceProperties(changes, RESOURCE, zone)
+
+
+class _Zone(object):
+
+    def __init__(self, client=None, project=TestChanges.PROJECT,
+                 name=TestChanges.ZONE_NAME):
+        self._client = client
+        self.project = project
+        self.name = name
+
+
+class _Client(object):
+
+    def __init__(self, project='project', connection=None):
+        self.project = project
+        self.connection = connection
+
+
+class _Connection(object):
```

Comment:
If so, it would be another cross-API cleanup:  we use that pattern frequently:

```bash
$ git grep -l _Client
gcloud/bigquery/test_dataset.py
gcloud/bigquery/test_job.py
gcloud/bigquery/test_table.py
gcloud/datastore/test_batch.py
gcloud/datastore/test_query.py
gcloud/datastore/test_transaction.py
gcloud/dns/test_changes.py
gcloud/dns/test_zone.py
gcloud/pubsub/test_subscription.py
gcloud/pubsub/test_topic.py
gcloud/resource_manager/test_project.py
gcloud/storage/test__helpers.py
gcloud/storage/test_acl.py
gcloud/storage/test_batch.py
gcloud/storage/test_blob.py
gcloud/storage/test_bucket.py
gcloud/storage/test_client.py
gcloud/test_credentials.py
gcloud/test_iterator.py
$ git grep -l _Connection
gcloud/bigquery/test_client.py
gcloud/bigquery/test_dataset.py
gcloud/bigquery/test_job.py
gcloud/bigquery/test_table.py
gcloud/datastore/test_batch.py
gcloud/datastore/test_helpers.py
gcloud/datastore/test_query.py
gcloud/datastore/test_transaction.py
gcloud/dns/test_changes.py
gcloud/dns/test_client.py
gcloud/dns/test_zone.py
gcloud/pubsub/test_client.py
gcloud/pubsub/test_subscription.py
gcloud/pubsub/test_topic.py
gcloud/resource_manager/test_client.py
gcloud/resource_manager/test_project.py
gcloud/storage/test__helpers.py
gcloud/storage/test_acl.py
gcloud/storage/test_batch.py
gcloud/storage/test_blob.py
gcloud/storage/test_bucket.py
gcloud/test_iterator.py
```